webapp/src/Controller/Team/SubmissionController.php

```diff
@@ -60,47 +61,135 @@ public function createAction(Request $request, ?Problem $problem = null): Respon
+        $formUpload->handleRequest($request);
+        $formPaste->handleRequest($request);
+        if ($formUpload->isSubmitted() || $formPaste->isSubmitted()) {
             if ($contest === null) {
                 $this->addFlash('danger', 'No active contest');
             } elseif (!$this->dj->checkrole('jury') && !$contest->getFreezeData()->started()) {
                 $this->addFlash('danger', 'Contest has not yet started');
             } else {
-                /** @var Problem $problem */
-                $problem = $form->get('problem')->getData();
-                /** @var Language $language */
-                $language = $form->get('language')->getData();
-                /** @var UploadedFile[] $files */
-                $files      = $form->get('code')->getData();
-                if (!is_array($files)) {
-                    $files = [$files];
+                $problem = null;
+                $language = null;
+                $files = [];
+                $entryPoint = null;
+                $message = '';
+        
+                if ($formUpload->isSubmitted() && $formUpload->isValid()) {
+                    $problem = $formUpload->get('problem')->getData();
+                    $language = $formUpload->get('language')->getData();
+                    $files = $formUpload->get('code')->getData();
+                    if (!is_array($files)) {
+                        $files = [$files];
+                    }
+                    $entryPoint = $formUpload->get('entry_point')->getData() ?: null;
+                } elseif ($formPaste->isSubmitted() && $formPaste->isValid()) {
+                    $problem = $formPaste->get('problem')->getData();
+                    $language = $formPaste->get('language')->getData();
+                    $codeContent = $formPaste->get('code_content')->getData();
+        
+                    if ($codeContent == null || empty(trim($codeContent))) {
+                        $this->addFlash('danger', 'No code content provided.');
+                        return $this->redirectToRoute('team_index');
+                    }
+        
+                    $tempDir = sys_get_temp_dir();
+                    $tempFileName = sprintf(
+                        'submission_%s_%s_%s.%s',
+                        $user->getUsername(),
+                        $problem->getName(),
+                        date('Y-m-d_H-i-s'),
+                        $language->getExtensions()[0]
+                    );
+                    $tempFileName = preg_replace('/[^a-zA-Z0-9_.-]/', '_', $tempFileName);
+
+                    if ($language->getExtensions()[0] == 'java' || $language->getExtensions()[0] == 'kt') {
+                        $entryPoint = $formPaste->get('entry_point')->getData() ?: null;
+                        // Check for invalid characters in entry point name
+                        $invalidChars = '/[<>:"\/\\|?*]/';
+                        if(preg_match($invalidChars, $entryPoint)) {
```

Comment:
There are some warnings on https://github.com/DOMjudge/domjudge/pull/2720/files which you should still handle.

Reply:
done

Reply:
I still see them:
<img width="1454" alt="image" src="https://github.com/user-attachments/assets/b43b358a-7ab5-410d-9c3e-7fe921f3af1f">

maybe you have to toggle `a` to see them.